src/seq_mv/csr_matvec_device.c

```diff
@@ -29,19 +37,45 @@ hypre_CSRMatrixMatvecDevice2( HYPRE_Int        trans,
                               hypre_Vector    *y,
                               HYPRE_Int        offset )
 {
+   /* Sanity check */
    if (hypre_VectorData(x) == hypre_VectorData(y))
    {
       hypre_error_w_msg(HYPRE_ERROR_GENERIC,
                         "ERROR::x and y are the same pointer in hypre_CSRMatrixMatvecDevice2");
    }
 
-#if defined(HYPRE_USING_CUSPARSE) || defined(HYPRE_USING_ROCSPARSE) || defined(HYPRE_USING_ONEMKLSPARSE)
-   if ( hypre_HandleSpMVUseVendor(hypre_handle()) )
+#if defined(HYPRE_USING_CUSPARSE)  || \
+    defined(HYPRE_USING_ROCSPARSE) || \
+    defined(HYPRE_USING_ONEMKLSPARSE)
+
+   /* Input variables */
+   HYPRE_Int  num_vectors_x      = hypre_VectorNumVectors(x);
+   HYPRE_Int  num_vectors_y      = hypre_VectorNumVectors(y);
+   HYPRE_Int  multivec_storage_x = hypre_VectorMultiVecStorageMethod(x);
+   HYPRE_Int  multivec_storage_y = hypre_VectorMultiVecStorageMethod(y);
+
+   /* Local variables */
+   HYPRE_Int  use_vendor;
+
+   /* Force use of hypre's SpMV for row-wise multivectors */
+   if ((num_vectors_x > 1 && multivec_storage_x == 1) ||
```

Comment:
It seems that `CUSPARSE_SPMM_CSR_ALG2` supports row-major?  https://docs.nvidia.com/cuda/cusparse/index.html#cusparse-generic-function-spmm

Reply:
Yes! I can add that. However when `(multivec_storage_x == 1 && multivec_storage_y == 0) || (multivec_storage_x == 0 && multivec_storage_y == 1)`, we'll still need to use force use of hypre's SpMV

Reply:
Where is `(multivec_storage_x == 1 && multivec_storage_y == 0) || (multivec_storage_x == 0 && multivec_storage_y == 1)` implemented? I am seeing whenever `multivec_storage_x/y == 1`, do not use vendor... I may be wrong.

Reply:
Correct! `(multivec_storage_x == 1 && multivec_storage_y == 0) || (multivec_storage_x == 0 && multivec_storage_y == 1)` would be a subset of what I currently have with this `if` statement.

Reply:
OK. Thanks! I think we would like to have the option of using vendor's kernel as long as we can, since they are constantly making improvement.

Reply:
Yes, makes sense!